classes/local/report/object_location_history_table.php

```diff
@@ -0,0 +1,138 @@
+class object_location_history_table extends \table_sql {
+
+    /**
+     * Constructor for the file location history table.
+     */
+    public function __construct() {
+        parent::__construct('locationhistory');
+
+        // TODO: Replace by lang strings.
+        $columnheaders = [
+            'date' => get_string('date'),
+            'local_count' => get_string('object_status:location:localcount', 'tool_objectfs'),
+            'local_size' => get_string('object_status:location:localsize', 'tool_objectfs'),
+            'duplicated_count' => get_string('object_status:location:duplicatedcount', 'tool_objectfs'),
+            'duplicated_size' => get_string('object_status:location:duplicatedsize', 'tool_objectfs'),
+            'external_count' => get_string('object_status:location:externalcount', 'tool_objectfs'),
+            'external_size' => get_string('object_status:location:externalsize', 'tool_objectfs'),
+            'missing_count' => get_string('object_status:location:missingcount', 'tool_objectfs'),
+            'missing_size' => get_string('object_status:location:missingsize', 'tool_objectfs'),
+            'total_count' => get_string('object_status:location:totalcount', 'tool_objectfs'),
+            'total_size' => get_string('object_status:location:totalsize', 'tool_objectfs'),
+            'filedir_count' => get_string('object_status:location:filedircount', 'tool_objectfs'),
+            'filedir_size' => get_string('object_status:location:filedirsize', 'tool_objectfs'),
+            'delta_count' => get_string('object_status:location:deltacount', 'tool_objectfs'),
+            'delta_size' => get_string('object_status:location:deltasize', 'tool_objectfs'),
+        ];
+
+        $this->is_downloadable(true);
+        $this->define_columns(array_keys($columnheaders));
+        $this->define_headers(array_values($columnheaders));
+        $this->define_header_column('date');
+        $this->collapsible(false);
+        $this->sortable(false);
+        $this->pageable(false);
+    }
+
+    /**
+     * Query the db. Store results in the table object for use by build_table.
+     *
+     * @param int $pagesize size of page for paginated displayed table.
+     * @param bool $useinitialsbar do you want to use the initials bar. Bar
+     * will only be used if there is a fullname column defined for the table.
+     * @throws \dml_exception
+     */
+    public function query_db($pagesize, $useinitialsbar = true) {
+        global $DB;
+        $sql = 'SELECT CONCAT(timecreated, datakey) AS uid,
+                       timecreated AS date,
+                       datakey AS location,
+                       objectcount AS count,
+                       objectsum AS size
+                  FROM {tool_objectfs_reports}
+                 WHERE reporttype = :reporttype
+              ORDER BY timecreated';
+        $params = array('reporttype' => 'location');
+        $rawrecords = $DB->get_records_sql($sql, $params, 0, 5000);
```

Comment:
this is probably better as a recordset

Reply:
Done. get_records is used

Reply:
This isn't done at all, it is still sing get_records, not get_recordset. This is a much deeper change